MediaBrowser.Controller/Net/BasePeriodicWebSocketListener.cs

```diff
@@ -25,11 +26,19 @@ public abstract class BasePeriodicWebSocketListener<TReturnDataType, TStateType>
         where TStateType : WebSocketListenerState, new()
         where TReturnDataType : class
     {
+        private readonly Channel<bool> _channel = Channel.CreateUnbounded<bool>(new UnboundedChannelOptions
+        {
+            AllowSynchronousContinuations = false,
+            SingleReader = true,
+            SingleWriter = false
+        });
+
+        private readonly SemaphoreSlim _lock = new(1, 1);
```

Comment:
Why not using the task from `HandleMessages` to await in the Dispose method?

Reply:
I forgot to close the channel in dispose